resources/lib/telstra_auth.py

```diff
@@ -0,0 +1,86 @@
+#-------------------------------------------------------------------------------
+import requests
+import collections
+import json
+import urlparse
+import config
+import comm
+import utils
+from exception import AFLVideoException
+
+class SortedHTTPAdapter(requests.adapters.HTTPAdapter):
+    def add_headers(self, request, **kwargs):
+        if 'header_order' in globals():
+            header_list = request.headers.items()
+            for item in header_list: 
+                if item[0] not in header_order:
+                    header_list.remove(item)
+            request.headers = collections.OrderedDict(
+                sorted(header_list, key=lambda x: header_order.index(x[0])))
+
+        
+def get_token(username, password, phone_number):
+    """ Obtain a valid token from Telstra, will be used to make requests for 
+        Ooyala embed tokens"""
+    session = requests.Session()
+    session.verify = False
+    session.mount("https://", SortedHTTPAdapter())
+    global header_order
+        
+    # Send our first login request to AFL API, recieve (unactivated) token
+    # and 'msisdn' URL
+
+    session.headers = {'x-media-mis-token': comm.fetch_token()}
+    auth_resp = session.post('http://api.sub.afl.com.au/cfs-premium/users?paymentMethod=ONE_PLACE')
+    jsondata = json.loads(auth_resp.text)
+    token = jsondata.get('uuid')
+    spurl = config.SPORTSPASS_URL.format(token)
+    session.headers = config.SPORTSPASS_HEADERS
+    sp = session.get(spurl)
+    msisdn_url = sp.url
+    
+    # Sign in to telstra.com to recieve cookies, get the SAML auth, and 
+    # modify the escape characters so we can send it back later
+    session.headers = config.SIGNON_HEADERS
+    signon_data = config.SIGNON_DATA
+    signon_data.update({'username': username, 'password': password})
+    signon = session.post(config.SIGNON_URL, data=signon_data)
+    utils.log(signon.url)
+    signon_query = dict(urlparse.parse_qsl(urlparse.urlsplit(signon.url)[3]))
+    if 'errorcode' in signon_query:
+        if signon_query['errorcode'] == '0':
+             raise AFLVideoException('Please enter your username in the settings')
+        if signon_query['errorcode'] == '1':
+             raise AFLVideoException('Please enter your password in the settings')
+        if signon_query['errorcode'] == '2':
+             raise AFLVideoException('Please enter your username and password in the settings')
+        if signon_query['errorcode'] == '3':
+             raise AFLVideoException('Please check your username and password in the settings')
+    start_index = signon.text.find('SAMLResponse')+21
```

Comment:
magic numbers are scary. is there a saml library we should use for this, or not worth it?